### PR TITLE
Scalenode terrain

### PR DESCRIPTION
**Pull request by mjtorn**
Because the gradient approach is hard to deal with and slow in its implementation, have an alternative to use lerping instead.

### CODE REVIEW

**Comment by mjtorn** [comment]
I think the main thing missing is documentation, but plz @StraToN check this out. I wrote it in my typical deadline-induced panic so there might be problems I just don't see.

At least our player moves smoothly and it should perform at least twice as good as the gradient scalemap thing.

File: device/globals/terrain_scalenodes.gd [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	update()
+
+static func sort_by_y(a, b):
+	return a.global_position.y < b.global_position.y
+
+# Return a "scale range" immediately based on the interpolated scale size
+func get_terrain(pos):
+	# printt("Called", pos)
+	var distant_color = Color(1, 1, 0, 1)
```
Comment (mjtorn) [inline device/globals/terrain_scalenodes.gd]: Please disregard this. I will force-push it out later.

**Comment by mjtorn** [comment]
Force-pushed removal of debug print

**Comment by StraToN** [comment]
I had developed a similar approach in my pre-Escoria demo, so I definitely agree with the principle. It simply misses documentation, other than that it's cool, thanks!

**Comment by mjtorn** [comment]
Thanks @StraToN ! I'm merging now with some documentation, so it doesn't get in the way of the next big thing :)

**Pull request merged**